fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/query/FDBNestedFieldQueryTest.java

```diff
@@ -274,18 +286,24 @@ public void nestedWithAnd() throws Exception {
                 ))
                 .build();
         plan = planner.plan(query);
-        assertThat(plan, filter(Query.field("stats").matches(
-                Query.and(Query.field("school_name").lessThan("University of Procrastination"),
-                        Query.field("hometown").startsWith("H"))),
-                indexScan(allOf(indexName("stats$school"), bounds(hasTupleString("([null],[1000]]"))))));
         if (planner instanceof RecordQueryPlanner) {
+            assertThat(plan, filter(Query.field("stats").matches(
+                    Query.and(Query.field("school_name").lessThan("University of Procrastination"),
+                            Query.field("hometown").startsWith("H"))),
+                    indexScan(allOf(indexName("stats$school"), bounds(hasTupleString("([null],[1000]]"))))));
+
             assertEquals(1700959433, plan.planHash(PlanHashable.PlanHashKind.LEGACY));
             assertEquals(1026881662, plan.planHash(PlanHashable.PlanHashKind.FOR_CONTINUATION));
             assertEquals(129783739, plan.planHash(PlanHashable.PlanHashKind.STRUCTURAL_WITHOUT_LITERALS));
         } else {
-            assertEquals(-1198378902, plan.planHash(PlanHashable.PlanHashKind.LEGACY));
-            assertEquals(-699838672, plan.planHash(PlanHashable.PlanHashKind.FOR_CONTINUATION));
-            assertEquals(-1395185561, plan.planHash(PlanHashable.PlanHashKind.STRUCTURAL_WITHOUT_LITERALS));
+            assertThat(plan, filter(
+                    allOf(queryPredicateDescendant(PredicateMatchers.field("stats", "school_name").lessThan("University of Procrastination")),
+                            queryPredicateDescendant(PredicateMatchers.field("stats", "hometown").startsWith("H"))),
```

Comment:
Three things can happen:
1. predicates may get reordered
2. predicates in the old planner are based on `QueryComponents`, in the new planner they are not.
3. there is an actual different plan getting produced. if that happens the plan the new planner produces is better as this would be a regression otherwise